### PR TITLE
Compile with -qstackprotect on xlc

### PR DESCRIPTION
**Pull request by pshipton**
Signed-off-by: Peter Shipton <Peter_Shipton@ca.ibm.com>

### CODE REVIEW

**Comment by pshipton** [comment]
jenkins test sanity aix jdk11

**Comment by pshipton** [comment]
I'll roll this out to the remaining versions after review.

**Review by keithc-ca — APPROVED** [review]

**Comment by keithc-ca** [comment]
> I'll roll this out to the remaining versions after review.

It's already present in jdk17 and newer versions:
* https://github.com/ibmruntimes/openj9-openjdk-jdk17/blob/a3ace8a9b67c0f38fc2eebf17a80e3f10e86f520/make/autoconf/flags-cflags.m4#L495
* https://github.com/ibmruntimes/openj9-openjdk-jdk18/blob/742164f41ccbae087b64a8f282ff98d610ab3f14/make/autoconf/flags-cflags.m4#L495
* https://github.com/ibmruntimes/openj9-openjdk-jdk18/blob/742164f41ccbae087b64a8f282ff98d610ab3f14/make/autoconf/flags-cflags.m4#L495

Each of those versions also include `-qstackprotect` in `TOOLCHAIN_CFLAGS_JVM`. Perhaps we should do likewise (even though `TOOLCHAIN_CFLAGS_JVM` seems only relevant to building Hotspot)?

**Comment by pshipton** [comment]
> It's already present in jdk17 and newer versions

Glad I was on the right track.

> even though TOOLCHAIN_CFLAGS_JVM seems only relevant to building Hotspot

That's what I figured. I've added it since it doesn't hurt.

File: make/autoconf/flags-cflags.m4 [hunk 1]
```diff
@@ -528,10 +532,10 @@ AC_DEFUN([FLAGS_SETUP_CFLAGS_HELPER],
     fi
 
   elif test "x$TOOLCHAIN_TYPE" = xxlc; then
-    TOOLCHAIN_CFLAGS_JDK="-qchars=signed -qfullpath -qsaveopt"  # add on both CFLAGS
+    TOOLCHAIN_CFLAGS_JDK="-qchars=signed -qfullpath -qsaveopt -qstackprotect"  # add on both CFLAGS
     TOOLCHAIN_CFLAGS_JVM="-qtune=balanced \
         -qalias=noansi -qstrict -qtls=default -qlanglvl=c99vla \
-        -qlanglvl=noredefmac -qnortti -qnoeh -qignerrno"
+        -qlanglvl=noredefmac -qnortti -qnoeh -qignerrno  -qstackprotect"
```
Comment (keithc-ca) [inline make/autoconf/flags-cflags.m4:538]: nit: extra space before `-qstackprotect`

Comment (pshipton) [reply make/autoconf/flags-cflags.m4]: Removed it.

**Comment by keithc-ca** [comment]
I think the [successful build](https://openj9-jenkins.osuosl.org/view/Pull%20Requests/job/PullRequest-OpenJDK11/42/) is sufficient testing of this.

**Pull request merged**